src/controllers/commands_controller.js

```diff
@@ -32,153 +32,101 @@
-                white: '#FCFBEA',
-                blue: '#1100FF',
-                purple: '#971AC4',
-                magenta: '#75009F',
-                pink: '#FFC0CB',
-              };
-              if (!colorname_to_hue[payload.param3]) {
-                response.status(404).json({message: 'Hue color not found'});
-                return;
-              } else {
-                iotOptions.body = JSON.stringify({color:
-                 colorname_to_hue[payload.param3]});
-                payload.href = obj.properties.color.href;
-              }
-            } else {
-              response.status(404).json({message: 'Command not found'});
-              return;
-            }
-            const iotUrl = CommandsController.gatewayHref + payload.href;
-            iotOptions.headers.Authorization = request.headers.authorization;
-            // Returning 201 to signify that the command was mapped to an
-            // intent and matched a 'thing' in our list. Return a response to
-            // caller with this status before the command finishes execution
-            // as the execution can take some time (e.g. blinds)
-            response.status(201).json({
-              message: 'Command Created',
-              payload: payload,
-            });
-            fetch(iotUrl, iotOptions)
-              .then(function() {
-                // In the future we may want to use WS to give a status of
-                // the disposition of the command execution..
-              })
-              .catch(function(err) {
-                // Future, give status via WS.
-                console.log(`catch inside PUT:${err}`);
-              });
-          } else {
-            response.status(404).json({message: 'Thing not found'});
-          }
-        }).catch(function(error) {
-          console.log('Error parsing intent:', error);
-          response.status(404).json({message:
-            'Internal error determining intent'});
-        });
-      }).catch(function(error) {
-        console.log('Error training:', error);
-        response.status(404).json({message:
-          'Internal error determining intent'});
-      });
-    }).catch(function(err) {
-      console.log(`error catch:${err}`);
-      response.status(500).send({
-        message: err,
-      });
+  if (thing) {
+    let propertyName, value;
+
+    if (payload.param2 === 'on' || payload.param2 === 'off') {
+      propertyName = 'on';
+      value = payload.param2 === 'on';
+    } else if ((payload.param2 === 'warmer' ||
+                payload.param2 === 'cooler') &&
+               thing.properties.colorTemperature) {
+      let current;
+      try {
+        current = await AddonManager.getProperty(thing.id, 'colorTemperature');
+      } catch (e) {
+        response.status(400).json({message: 'Failed to set property'});
+        return;
+      }
+
+      propertyName = 'colorTemperature';
+      value = payload.param2 === 'warmer' ? current - 100 : current + 100;
```

Comment:
These adjustment values may need to be tweaked a bit, not sure what others think.